src/Form/MetadataDisplayForm.php

```diff
@@ -11,69 +19,205 @@
-    /** @var \Drupal\Core\Template\TwigEnvironment $environment */
-    $environment = \Drupal::service('twig');
-    $environment->invalidate();
+    $this->formatterPluginManager->clearCachedDefinitions();
+    $this->twig->invalidate();
 
-    $form_state->setRedirectUrl($this->entity->toUrl('collection'));
     return $status;
   }
+
+  /**
+   * AJAX callback.
+   */
+  public static function ajaxPreview($form, FormStateInterface $form_state) {
+    $response = new AjaxResponse();
+
+    /** @var \Drupal\format_strawberryfield\MetadataDisplayInterface $entity */
+    $entity = $form_state->getFormObject()->getEntity();
+
+    // Attach the library necessary for using the OpenOffCanvasDialogCommand and
+    // set the attachments for this Ajax response.
+    $form['#attached']['library'][] = 'core/drupal.dialog.off_canvas';
+    $form['#attached']['library'][] = 'codemirror_editor/editor';
+    $response->setAttachments($form['#attached']);
+
+    $input = $form_state->getUserInput();
+    $id = EntityAutocomplete::extractEntityIdFromAutocompleteInput($input['ado_context_preview']);
+    /** @var \Drupal\node\NodeInterface $preview_node */
+    $preview_node = \Drupal::entityTypeManager()
+      ->getStorage('node')
+      ->load($id);
+    if (empty($preview_node)) {
+      return $response;
+    }
+
+    $sbf_fields = \Drupal::service('strawberryfield.utility')->bearsStrawberryfield($preview_node);
+
+    // Set initial context.
+    $context = [
+      'node' => $preview_node,
+      'iiif_server' => \Drupal::service('config.factory')->get('format_strawberryfield.iiif_settings')->get('pub_server_url'),
+    ];
+
+    // Add the SBF json context.
+    // @see MetadataExposeDisplayController::castViaTwig()
+    foreach ($sbf_fields as $field_name) {
+      /** @var \Drupal\strawberryfield\Field\StrawberryFieldItemList $field */
+      $field = $preview_node->get($field_name);
+      foreach ($field as $offset => $fielditem) {
+        $jsondata = json_decode($fielditem->value, TRUE);
+        // Preorder as:media by sequence.
+        $ordersubkey = 'sequence';
+        foreach (StrawberryfieldJsonHelper::AS_FILE_TYPE as $key) {
+          StrawberryfieldJsonHelper::orderSequence($jsondata, $key, $ordersubkey);
+        }
+        if ($offset === 0) {
+          $context['data'] = $jsondata;
+        }
+        else {
+          $context['data'][$offset] = $jsondata;
+        }
+      }
+    }
+    // Ensure we're using the twig from user's input instead of the entity's
+    // default.
+    $entity->set('twig', $input['twig'][0]);
+    $output = [
+      '#type' => 'codemirror',
+      '#rows' => 60,
+      '#value' => $entity->renderNative($context),
+      '#codemirror' => [
+        'lineNumbers' => FALSE,
+        'toolbar' => FALSE,
+        'readOnly' => TRUE,
+        'mode' => 'text/html',
+      ],
+    ];
+
+    $response->addCommand(new OpenOffCanvasDialogCommand(t('Preview'), $output, ['rezisable' => true, 'width' => '50%']));
```

Comment:
This is a typo:
Should be  'resizable' according to https://api.drupal.org/api/drupal/core%21modules%21outside_in%21src%21Ajax%21OpenOffCanvasDialogCommand.php/class/OpenOffCanvasDialogCommand/8.2.x

Reply:
I guess it is the default :)

Reply:
Since you have `rezisable` instead of `resizable`, you can either remove it (and let the default take over) or fix the typo. Up to you

Reply:
Pushed the fix

Reply:
Thanks!